wear/src/main/java/io/homeassistant/companion/android/home/HomeActivity.kt

```diff
@@ -103,10 +119,11 @@ class HomeActivity : ComponentActivity(), HomeView {
         finish()
     }
 
+    @ExperimentalWearMaterialApi
     @Composable
-    private fun LoadHomePage(entities: Array<Entity<Any>>) {
+    private fun LoadHomePage(entities: Array<Entity<Any>>, favorites: MutableSet<String>) {
 
-        if (entities.isNullOrEmpty()) {
+        if (entities.isNullOrEmpty() && favorites.isNullOrEmpty()) {
```

Comment:
Is there any possibility where entities is null or empty, but favorites is not?

Reply:
Yes there is, so down below in the `else` we do individual checks for favorites and entities again. So basically if the user has not had entities loaded yet and hasnt set favorites they see old loading. If they dont have entities yet but do have favorites they will see favorites and the loading chip, then loading screen/chip will disappear once we get the list of entities.